clients/src/test/java/org/apache/kafka/common/utils/MockTime.java

```diff
@@ -24,6 +25,15 @@
  */
 public class MockTime implements Time {
 
+    interface MockTimeListener {
+        void tick();
+    }
+
+    /**
+     * Listeners which are waiting for time changes.
+     */
+    private final CopyOnWriteArrayList<MockTimeListener> listeners = new CopyOnWriteArrayList<>();
```

Comment:
Is there a reason for this to be a `CopyOnWriteArrayList` as opposed to say a `ConcurrentLinkedQueue`?

Reply:
I would expect that an arraylist would perform slightly better than a linked list / queue when there were multiple elements.  However, the difference is small, especially since we currently only have one element.